src/patterns.rs

```diff
@@ -199,18 +195,19 @@ impl Patterns {
     }
 }
 
-/// Parses a string like "3-5", "-5", or "3-" into min and max length values.
-/// Returns `Some((min, max))` where each is an `Option<usize>`.
+/// Parses a string like "3-5", "-5", "3-", or "3" into min and max length values.
+/// Returns `Some((min, max))` where each is an `Option<usize>` unless the input is empty, in which
+/// case it returns `None`.
 fn parse_length_range(input: &str) -> Option<(Option<usize>, Option<usize>)> {
     if input.is_empty() {
         return None;
     }
     let parts: Vec<&str> = input.split('-').collect();
+    if parts.len() > 2 { // TODO? return error instead?
```

Comment:
I assume that we want to disallow things with more than 1 `-`

Reply:
yep